### PR TITLE
feat: add erathostene algorithm

### PR DESCRIPTION
**Pull request by GabinDDL**
add erathostene algorithm

### CODE REVIEW

File: content/number-theory/FastEratosthenes.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"""
+ Author: Gabin Dudillieu
```
Comment (Yag000) [inline content/number-theory/FastEratosthenes.py]: Bad comment format, see : https://github.com/UPCGM-ICPC/UPCGM-Notebook/pull/6/files

File: content/number-theory/FastEratosthenes.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ Date: 2024-01-26
+ Description: Prime sieve for generating all primes smaller than LIM.
+ Time: LIM=1e9 $\approx$ 1.5s
+ Status: Stress-tested
+ Details: Despite its n log log n complexity, segmented sieve is still faster
+ than other options, including bitset sieves and linear sieves. This is
+ primarily due to its low memory usage, which reduces cache misses. This
+ implementation skips even numbers.
```
Comment (Yag000) [inline content/number-theory/FastEratosthenes.py]: I'm afraid I cannot see how that is being done.

File: content/number-theory/FastEratosthenes.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+"""
+ Author: Gabin Dudillieu
+ Date: 2024-01-26
+ Description: Prime sieve for generating all primes smaller than LIM.
+ Time: LIM=1e9 $\approx$ 1.5s
+ Status: Stress-tested
```
Comment (Yag000) [inline content/number-theory/FastEratosthenes.py]: Where?

**Review by Yag000 — CHANGES_REQUESTED** [review]

**Comment by GabinDDL** [comment]
@Yag000 normally it's resolved

File: content/number-theory/FastEratosthenes.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+"""
+* Author: Gabin Dudillieu
+* Date: 2024-01-26
+* Description: Prime sieve for generating all primes smaller than LIM.
+* Time: LIM=1e9 $\approx$ 1.5s
```
Comment (Yag000) [inline content/number-theory/FastEratosthenes.py]: I cannot reproduce these times, I'm at 17s with only 1e8 and 5-10s for 1e7

Comment (GabinDDL) [reply content/number-theory/FastEratosthenes.py]: yes you're right, I replaced it by the complexity

**Comment by Yag000** [comment]
As for #13 ci is not passing due to tests

**Review by Yag000 — APPROVED** [review]

File: content/number-theory/FastEratosthenes.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    p = 2
+    while (p * p <= num):
+        if (prime[p] == True):
+            for i in range(p * p, num+1, p):
+                prime[i] = False
+        p += 1
+    for p in range(2, num+1):
+        if prime[p]:
+            result.append(p)
+    return result
```
Comment (YanisLcn) [inline content/number-theory/FastEratosthenes.py:19]: If needed.
```suggestion
return [i for i, e in enumerate(prime) if i >= 2 and e]
```

**Review by YanisLcn — APPROVED** [review]

**Pull request merged**